src/useDebouncedCallback.ts

```diff
@@ -71,11 +71,6 @@ function useDebouncedCallback<TCallback extends (...args: any[]) => any>(
 
   const timeout = useTimeout()
 
-  useWillUnmount(() => {
-    clearTimeout(timerRef.current)
-    isTimerSetRef.current = false
```

Comment:
I think this is ok to remove since the only time a pending timeout would be left `true` is when it's unmounted

Reply:
I think we do need to reset `isTimerSetRef`, since on remount, it restores the previous state of the component.  I did run into this in my stuff where it wouldn't invoke the callback.

I can build this and double check it at work on Monday.

Reply:
I think that was true due to useTimeout not being strictly speaking StrictMode safe. If the component is actually unmounted and remounted the ref will be reset, but _not_ during the double useEffect call of StrictMode. I think since this is now never changes in an effect it should be true past unmounting but I think that's ok since it only matters if the timers are still running past mount. IDK I couldn't really think of a good way to reset this without `useWillUnmount` which is going to set this to `false` immediately

Reply:
Tested with the lib I had at work and it works fine.  LGTM!